tests/richText.js

```diff
@@ -345,73 +329,50 @@ define([
             var output = '<output value="/data/text" />';
             editor.setValue('one two', function () {
                 assert.equal(editor.getValue(), 'one two');
-                select(editor, 3);
+                editor.select(3);
                 editor.insertOutput(output);
                 assert.equal(editor.getValue(), "one" + output + " two");
                 done();
             });
         });
 
-        // TODO tests to make sure select() works in various scenarios
-        // for example, with multiple lines
-
-        // -- helpers ---------------------------------------------------------
+        function applyArgs(func) {
+            return function (args) {
+                return func.apply(this, args);
+            };
+        }
 
-        function select(editor, start) {
-            function iterNodes(element) {
-                var i = 0,
-                    children = element.getChildren(),
-                    count = children.count(),
-                    inner = null;
-                function next() {
-                    var child;
-                    if (inner) {
-                        child = inner();
-                        if (child !== null) {
-                            return child;
-                        }
-                        inner = null;
+        _.each([
+            ["one two", 3, "one/data/text two"],
+            ["one two", 4, "one /data/text two"],
+            ["one\n\ntwo", 3, "one/data/text\n\ntwo"],
+            ["one\n\ntwo", 4, "one\n/data/text\ntwo"],
+            /* TODO make these tests pass
```

Comment:
Yes, they work in the UI. It's just that the programmatic selection logic gets confused beyond the first (or second?) line, probably because I'm counting nodes wrong in my selection logic. But that shouldn't affect any use of `ckSelect` right now since the index parameter is always zero in the only place it's called (outside of tests).